skimage/morphology/rolling_ball.py

```diff
@@ -0,0 +1,197 @@
+      canonical ellipsis equation
+      .. code-block:: python
+
+        | semi_spatial = kernel_shape / 2
+        | semi_vertex = intensity_vertex / 2
+        | np.sum((pos/semi_spatial)**2) + (intensity/semi_vertex)**2 = 1
+
+    - This algorithm assumes that dark pixels correspond to the background. If
+      you have a bright background, invert the image before passing it to the
+      function, e.g., using `utils.invert`.
+    - This algorithm is sensitive to noise (in particular salt-and-pepper
+      noise). If this is a problem in your image, you can apply mild
+      gaussian smoothing before passing the image to this function.
+
+    References
+    ----------
+    .. [1] Sternberg, Stanley R. "Biomedical image processing." Computer 1
+           (1983): 22-34. :DOI:`10.1109/MC.1983.1654163`
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from skimage import data
+    >>> from skimage.morphology import rolling_ellipsoid
+    >>> image = data.coins()
+    >>> background = rolling_ellipsoid(data.coins())
+    >>> filtered_image = image - background
+    """
+
+    if num_threads is None:
+        num_threads = 0
+
+    kernel_shape = np.asarray(kernel_shape)
+    if kernel_shape.ndim == 0:
+        kernel_shape = np.full_like(image.shape, kernel_shape)
+
+    kernel_shape_int = np.asarray(kernel_shape // 2 * 2 + 1, dtype=np.intp)
+
+    intensity_vertex = np.asarray(intensity_vertex, dtype=np.float_)
+    intensity_vertex = intensity_vertex / 2
+
+    image = np.asarray(image)
+    img = image.astype(np.float_)
+
+    ellipsoid_coords = np.stack(
+        np.meshgrid(
+            *[range(-x, x + 1) for x in kernel_shape_int // 2],
+            indexing='ij'
+        ),
+        axis=-1).reshape(-1, len(kernel_shape))
+    tmp = np.sum(
+        (ellipsoid_coords / (kernel_shape[np.newaxis, :] / 2)) ** 2, axis=1)
+    ellipsoid_intensity = intensity_vertex - \
+        intensity_vertex * np.sqrt(np.clip(1 - tmp, 0, None))
+    ellipsoid_intensity = ellipsoid_intensity.astype(np.float_)
+
+    pad_amount = np.round(kernel_shape / 2).astype(int)
+    img = np.pad(img, pad_amount[:, np.newaxis],
+                 constant_values=np.Inf, mode="constant")
+
+    kernel = np.where(tmp <= 1, 1, np.inf)
+
+    if has_nan:
+        background = apply_kernel_nan(
+            img.ravel(),
+            kernel,
+            ellipsoid_intensity,
+            np.array(image.shape, dtype=np.intp),
+            np.array(img.shape, dtype=np.intp),
+            kernel_shape_int,
+            num_threads
+        )
+    else:
+        background = apply_kernel(
+            img.ravel(),
+            kernel,
+            ellipsoid_intensity,
+            np.array(image.shape, dtype=np.intp),
+            np.array(img.shape, dtype=np.intp),
+            kernel_shape_int,
+            num_threads
+        )
+
+    background = background.astype(image.dtype)
+
+    return background
+
+
+def rolling_ball(image, radius=50, **kwargs):
```

Comment:
```suggestion
def rolling_ball(image, radius=50,has_nan=False, num_threads=None):
```
Can also please add the corresponding parameters' documentation and modify the `rolling_ellipsoid` call?

Reply:
I added the keyword arguments. Is it necessary to copy the documentation from `rolling_ellipsoid`, too? 

I added a reference in the `See Also` section because the [numpy docstring guide](https://numpydoc.readthedocs.io/en/latest/format.html) states

> [...] Routines whose docstrings further explain parameters used by this function are good candidates.

Not sure if I am doing this right though.